src/java/uk/ac/rdg/resc/edal/cdm/DataChunk.java

```diff
@@ -118,10 +118,11 @@ public Index getIndex() {
      * @return the data value, or {@link Float#NaN} if this is a missing value
      */
     public float readFloatValue(Index index) {
-        double val = arr.getDouble(index);
-        if (this.needsEnhance)
-            return (float) this.var.convertScaleOffsetMissing(val);
-        else
-            return (float) val;
+        double val = arr.getFloat(index);
```

Comment:
An explicit cast and a comment about the `VariableDS` methods would make me feel better, I think.